### DISCUSSION TITLE
Using custom Trivy image from private registry

### DISCUSSION THREAD

**Discussion by kfox1111** [aquasecurity/starboard#666] [Development]
You can override the image used with trivy with:
trivy.imageRef

But there is no way to specify trivy.imagePullSecret

**Comment by danielpacak** [comment]
The operator's pod and scan pods (e.g. Trivy scanner) use the `starboard-operator` service account. Hence, you can create an imagePullSecret and add it as a reference to the `starboard-operator` service account as explained in [Add ImagePullSecrets to a service account](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#add-imagepullsecrets-to-a-service-account).

I'm not sure it's a viable solution in your case, but we're trying to strike a balance between adding new configuration params and using existing K8s configuration facilities. Would that solution work for you @kfox1111 ? If we add trivy.imagePullSecret we'd probably have to do the same for all plugins.

**Reply by kfox1111** [reply]
Yeah. it could be solved by https://github.com/aquasecurity/starboard/discussions/646  and if the helm chart creates the service accounts, the option to specify imagePullSecrets in the config.

**Reply by kfox1111** [reply]
usability wise, that is a little bit twisty though, so you might get more questions about how to do that. If its in the trivy config right by imageRef, then its very obvious how to do it.
